src/lava/magma/runtime/runtime_service.py

```diff
@@ -16,6 +16,33 @@
     MGMT_COMMAND,
 )
 
+"""This file defines the interface for RuntimeService which is responsible for 
+coordinating the execution of a group of process models belonging to a common 
+synchronization domain. The domain might follow a SyncProtocol or could be 
+asynchronous too. The processes and their corresponding process models are 
+selected by the Runtime depending on the RunConfiguration assigned at the 
+start of execution. For each group of processes which follow the same 
+protocol and would execute on the same node, Runtime creates a RuntimeService
+which will coordinate all actions/commands from Runtime onto the processes as 
+well as return any acknowledgement back to Runtime.
+
+Currently we envision few different kinds of RuntimeService:
+
+1. PyRuntimeService: (Abstract Class) Coordinates process models executing in 
```

Comment:
We need to rethink this. I feel we should not have to write a new RuntimeService for each protocol. That might just be our current choice.

The "space" lingo sounds a bit wired. What is a CPU space? There are CPUs everywhere.